enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,439 @@
+add-on configuration, and it's not easy to locate the configuration object for add-on implementation.
+
+To make the `ConfigCoordinates` more flexible, we want to refactor the `ConfigCoordinates` to
+
+``` golang
+// ConfigCoordinates represents the information for locating the add-on configuration.
+type ConfigCoordinates struct {
+	// Deprecated: Use Group, Version and Resource fileds instead
+	// crdName is the name of the CRD used to configure instances of the managed add-on.
+	// This field should be configured if the add-on have a CRD that controls the configuration of the add-on.
+	// +optional
+	CRDName string `json:"crdName"`
+
+	// Deprecated: Use Group, Version and Resource fileds instead
+	// crName is the name of the CR used to configure instances of the managed add-on.
+	// This field should be configured if add-on CR have a consistent name across the all of the ManagedCluster instaces.
+	// +optional
+	CRName string `json:"crName"`
+
+	// group of the add-on configuration.
+	Group string `json:"group"`
+
+	// version of the add-on configuration.
+	Version string `json:"version"`
+
+	// resource of the add-on configuration.
+	Resource string `json:"resource"`
+
+	// namespace of the default add-on configuration.
+	// In scenario where all add-ons have a same configuration.
+	// +optional
+	DefaultNamespace string `json:"namespace,omitempty"`
+
+	// name of the default add-on configuration.
+	// In scenario where all add-ons have a same configuration.
+	// +optional
+	DefaultName string `json:"name,omitempty"`
+}
+```
+
+When the `ConfigCoordinates` is defined in the `ClusterManagementAddOnSpec`, the addon-framework will sync the
+information of `ConfigCoordinates` to the `ManagedClusterAddOn` status.
+
+In the `ManagedClusterAddOn` spec, we will introduce an add-on configuration referent, it is used to specify the add-on
+configuration namespace and name, and introduce a `ConfigReference` instead of the `ConfigCoordinates` in the
+`ManagedClusterAddOn` status
+
+```golang
+// ManagedClusterAddOnSpec defines the install configuration of an add-on agent on managed cluster.
+type ManagedClusterAddOnSpec struct {
+	// installNamespace is the namespace on the managed cluster to install the add-on agent.
+	// If it is not set, open-cluster-management-agent-addon namespace is used to install the add-on agent.
+	// +optional
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// configuration represents the configuration namespace and name for the current add-on.
+	// In scenario where the current add-on has its own configuration.
+	// +optional
+	Configuration ConfigurationReferent `json:"configuration,omitempty"`
+}
+
+// ConfigurationReferent represents the namespace and name for the current add-on.
+type ConfigurationReferent struct {
+	// namespace of the current add-on configuration.
+	// +optional
```

Comment:
we should clear state if it is not set, the related namespace/name in clustermanagementaddon is used if any?

Reply:
I clear this in the status part